src/main/java/de/tum/in/www1/artemis/service/quiz/QuizExerciseService.java

```diff
@@ -458,4 +458,38 @@ protected QuizExercise saveAndFlush(QuizExercise quizExercise) {
         log.debug("Save quiz exercise to database: {}", quizExercise);
         return quizExerciseRepository.saveAndFlush(quizExercise);
     }
+
+    /**
+     *
+     * @param newQuizExercise the newly created quiz exercise, after importing basis of imported exercise
+     * @param files           the new files to be added to the newQuizExercise which do not have a previous path and need to be saved in the server
+     * @return the new exercise with the updated file paths which have been created and saved
+     * @throws IOException
+     */
+    public QuizExercise uploadNewFilesToNewImportedQuiz(QuizExercise newQuizExercise, List<MultipartFile> files) throws IOException {
+        Map<String, MultipartFile> fileMap = files.stream().collect(Collectors.toMap(MultipartFile::getOriginalFilename, file -> file));
```

Comment:
```suggestion
        Map<String, MultipartFile> fileMap = files.stream().collect(Collectors.toMap(MultipartFile::getOriginalFilename, Function.identity()));
```
Makes it a bit clearer that the file stays unchanged